halo2_proofs/src/dev/failure.rs

```diff
@@ -411,18 +437,20 @@ fn render_lookup<F: Field>(
         FailureLocation::OutsideRegion { row } => *row,
     } as i32;
 
-    // Recover the fixed columns from the table expressions. We don't allow composite
-    // expressions for the table side of lookups.
+    // Recover the fixed columns from the table expressions.
+    // We don't allow composite expressions for the table side of fixed lookups.
+    // The table side of Dynamic lookups have a tag column.
     let table_columns = lookup.table_expressions.iter().map(|expr| {
         expr.evaluate(
             &|_| panic!("no constants in table expressions"),
             &|_| panic!("no selectors in table expressions"),
+            &|_| panic!("no virtual columns in table expressions"),
             &|query| format!("F{}", query.column_index),
-            &|_| panic!("no advice columns in table expressions"),
+            &|query| format!("A{}", query.column_index),
             &|_| panic!("no instance columns in table expressions"),
             &|_| panic!("no negations in table expressions"),
             &|_, _| panic!("no sums in table expressions"),
-            &|_, _| panic!("no products in table expressions"),
+            &|a, b| format!("{} * {}", a, b),
```

Comment:
Why are `products` now required? I presume because the Advice assigned into the `DynamicLookupTable` can come from any expression result.

If so, `sum`m `negation` and `scale` should be supported too.

Reply:
For fixed lookups, each table expression must be a fixed column.

In dynamic lookups, we will have a product between a `tag_column: Fixed` and a  `table_column: Advice`.